content/blog/2023/2024-03-15-getting-started-with-spring-security-and-jwt.md

```diff
@@ -0,0 +1,963 @@
+                                 JwtAuthenticationEntryPoint 
+                                             jwtAuthenticationEntryPoint) {
+
+        this.jwtFilter = jwtFilter;
+        this.authUserDetailsService = authUserDetailsService;
+        this.jwtAuthenticationEntryPoint = jwtAuthenticationEntryPoint;
+    }
+
+    @Bean
+    public DaoAuthenticationProvider authenticationProvider() {
+        final DaoAuthenticationProvider daoAuthenticationProvider = 
+                new DaoAuthenticationProvider();
+        daoAuthenticationProvider.setUserDetailsService(authUserDetailsService);
+        daoAuthenticationProvider.setPasswordEncoder(
+                PlainTextPasswordEncoder.getInstance());
+        return daoAuthenticationProvider;
+    }
+
+    @Bean
+    public AuthenticationManager authenticationManager(HttpSecurity httpSecurity) 
+            throws Exception {
+        return httpSecurity.getSharedObject(AuthenticationManagerBuilder.class)
+                .authenticationProvider(authenticationProvider())
+                .build();
+    }
+}
+````
+The `AuthenticationManager` uses the `AuthUserDetailsService` which uses the `spring.security.user` property.
+Now that we have the `AuthenticationManager` in place, let's look at how the `TokenService` is defined:
+````java
+@Service
+public class TokenService {
+
+    private final AuthenticationManager authenticationManager;
+
+    private final AuthUserDetailsService userDetailsService;
+
+    private final JwtHelper jwtHelper;
+
+    public TokenService(AuthenticationManager authenticationManager,
+                        AuthUserDetailsService userDetailsService,
+                        JwtHelper jwtHelper) {
+        this.authenticationManager = authenticationManager;
+        this.userDetailsService = userDetailsService;
+        this.jwtHelper = jwtHelper;
+    }
+
+
+    public TokenResponse generateToken(TokenRequest tokenRequest) {
+        this.authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        tokenRequest.getUsername(), tokenRequest.getPassword()));
+        final UserDetails userDetails = 
+                userDetailsService.loadUserByUsername(tokenRequest.getUsername());
+        String token = jwtHelper.createToken(
+                Collections.emptyMap(), userDetails.getUsername());
+        return TokenResponse.builder()
+                .token(token)
+                .build();
+    }
+}
+````
+`TokenResponse` is the Response object that contains the token string:
+````java
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class TokenResponse {
+
+    private String token;
+
+}
+````
+With the API now created, let's start our application and try to hit the endpoint using Postman.
+We see a `401 Unauthorized` error as below:
+{{% image alt="settings" src="images/posts/spring-security-and-jwt/PostmanToken-401.png" %}}
```

Comment:
Some of the screenshots have quite some white space at the bottom. Let's crop them to avoid having too much white space.

Reply:
Edited screenshots as applicable!